devicetypes/smartthings/cooper-scene-controller.src/cooper-scene-controller.groovy

```diff
@@ -0,0 +1,268 @@
+	cmds << new physicalgraph.device.HubAction(zwave.sceneControllerConfV1.sceneControllerConfGet(groupId:4).format())
+	cmds << new physicalgraph.device.HubAction(zwave.indicatorV1.indicatorSet(value: 15).format())
+	cmds << new physicalgraph.device.HubAction(zwave.associationV1.associationGet(groupingIdentifier:5).format())
+	cmds << new physicalgraph.device.HubAction(zwave.sceneControllerConfV1.sceneControllerConfGet(groupId:5).format())
+	cmds << new physicalgraph.device.HubAction(zwave.indicatorV1.indicatorSet(value: 31).format())
+	cmds << new physicalgraph.device.HubAction(zwave.indicatorV1.indicatorSet(value: 0).format())
+	//Make sure cloud is in sync with device
+	cmds << new physicalgraph.device.HubAction(zwave.indicatorV1.indicatorGet().format())
+	if (!getDataValue("manufacturer")) {
+		cmds << new physicalgraph.device.HubAction(zwave.manufacturerSpecificV2.manufacturerSpecificGet().format())
+	}
+	//Long interval to make it possible to process association set commands if necessary
+	sendHubCommand(cmds, 3000)
+}
+
+def refresh() {
+	def cmds = []
+	// when using All Off feature, controller will often not respond to Indicator Get command
+	// using delayed refresh proves more reliable in this case
+	cmds << new physicalgraph.device.HubAction("delay 500")
+	cmds << new physicalgraph.device.HubAction(zwave.indicatorV1.indicatorGet().format())
+	//Indicator returns number which is a bit representation of current state of switches
+	sendHubCommand(cmds, 500)
+}
+
+def parse(String description) {
+	if (description.startsWith("Err")) {
+		if (description.startsWith("Err 106") && !state.sec) {
+			state.sec = 0
+		}
+	} else {
+		def cmd = zwave.parse(description)
+		if (cmd) {
+			zwaveEvent(cmd)
+		}
+	}
+	return []
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.ManufacturerSpecificReport cmd) {
+	if (cmd.manufacturerName) {
+		updateDataValue("manufacturer", cmd.manufacturerName)
+	}
+	if (cmd.productTypeId) {
+		updateDataValue("productTypeId", cmd.productTypeId.toString())
+	}
+	// productId 0 is a valid value
+	if (cmd.productId != null) {
+		updateDataValue("productId", cmd.productId.toString())
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.associationv2.AssociationReport cmd) {
+	if (cmd.nodeId.any { it == zwaveHubNodeId }) {
+		sendEvent(descriptionText: "$device.displayName is associated in group ${cmd.groupingIdentifier}")
+	} else {
+		// We're not associated properly to this group, try setting association two times
+		def cmds = []
+		//Set Association for this group
+		cmds << new physicalgraph.device.HubAction(zwave.associationV1.associationSet(groupingIdentifier:cmd.groupingIdentifier, nodeId:[zwaveHubNodeId]).format())
+		cmds << new physicalgraph.device.HubAction(zwave.associationV1.associationSet(groupingIdentifier:cmd.groupingIdentifier, nodeId:[zwaveHubNodeId]).format())
+		sendHubCommand(cmds, 1500)
+	}
+}
+
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicSet cmd) {
```

Comment:
Seems strange to me that the device sends a basic set but never a basic report, but I don't have the docs to verify.

Reply:
Device can control scene capable devices.
In that case it sends SceneActivationSet with appropriate sceneId when switch is switched on, and BasicSet(value: 0) if switch is switched off. In this mode, device does not send any BasicReport nor does it respond to BasicGet commands.

For non-scene capable devices, it is possible to configure level for each association group for each switch. In that case we get BasicSet with value as configured when scene is activated, and BasicSet(value: 0) when scene is deactivated. In this mode, device does not send any BasicReport nor does it respond to BasicGet commands.

Reply:
We currently work on acccount st.dth.dev@gmail.com, location name US_LAB

We will leave device plugged in overnight. You can impersonate to check it if you want.

If you prefer we can collect telnet logs  I don't have any other ideas.